src/SignatureHelpProvider.php

```diff
@@ -0,0 +1,211 @@
+<?php
+declare(strict_types = 1);
+
+namespace LanguageServer;
+
+use PhpParser\ErrorHandler\Collecting;
+use PhpParser\Node;
+use LanguageServer\Index\ReadableIndex;
+use LanguageServer\Protocol\{
+    Range,
+    Position,
+    SignatureHelp,
+    SignatureInformation,
+    ParameterInformation
+};
+
+class SignatureHelpProvider
+{
+    /**
+     * @var DefinitionResolver
+     */
+    private $definitionResolver;
+
+    /**
+     * @var ReadableIndex
+     */
+    private $index;
+
+    /**
+     * @var Parser
+     */
+    private $parser;
+
+    /**
+     * @var Parser
+     */
+    private $parserErrorHandler;
+
+    /**
+     * @param DefinitionResolver $definitionResolver
+     * @param ReadableIndex      $index
+     */
+    public function __construct(DefinitionResolver $definitionResolver, ReadableIndex $index)
+    {
+        $this->definitionResolver = $definitionResolver;
+        $this->index = $index;
+        $this->parser = new Parser;
+        $this->parserErrorHandler = new Collecting;
+    }
+
+    /**
+     * Returns signature help for a specific cursor position in a document
+     *
+     * @param PhpDocument $doc The opened document
+     * @param Position $pos The cursor position
+     * @return SignatureHelp
+     */
+    public function provideSignature(PhpDocument $doc, Position $pos) : SignatureHelp
+    {
+        $help = new SignatureHelp;
+        $help->signatures = [];
+
+        $handle = fopen('php://temp', 'r+');
+        fwrite($handle, $doc->getContent());
+        fseek($handle, 0);
```

Comment:
The node at the requested position is `null`, after backtracking 31 symbols (each of those positions is still `null`) we get to an `Error` node, whose parent is a `PropertyFetch` node. The `$var` property of the fetch is populated, but the `$name` property is not (it is the `Error` node). This is why I used string operations to get the method being called, and then - the `Lexer` with some custom logic on top to get the parameter count. 
As I said in the Signature Help issue - it is a messy solution, but it is the only one I could come up with. I am sure someone with superior skills can take it a few steps further to something elegant and more robust :)